kore/src/Kore/Builtin/Int.hs

```diff
@@ -475,7 +475,8 @@ unifyInt term1 term2 unifyData =
 
 data UnifyIntEq = UnifyIntEq
     { eqTerm :: !(EqTerm (TermLike RewritingVariableName))
-    , value :: !Bool
+    , boolValue :: !Bool
+    , boolTerm :: !(TermLike RewritingVariableName)
```

Comment:
Why not replace both of these fields with an [`InternalBool`](https://github.com/kframework/kore/blob/221cb82050490c6b711e1e68b42390971e60685a/kore/src/Kore/Internal/InternalBool.hs#L26)? It is _the_ internal representation of Bool.